src/main/java/com/limechain/storage/block/BlockState.java

```diff
@@ -0,0 +1,796 @@
+package com.limechain.storage.block;
+
+import com.limechain.network.protocol.warp.dto.Block;
+import com.limechain.network.protocol.warp.dto.BlockBody;
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.runtime.Runtime;
+import com.limechain.storage.DBConstants;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.block.exception.LowerThanRootException;
+import com.limechain.storage.block.tree.BlockTree;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.Getter;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.SerializationUtils;
+import org.javatuples.Pair;
+
+import java.math.BigInteger;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * contains the historical block data of the blockchain, including block headers and bodies.
+ * It wraps the blocktree (which contains unfinalized blocks) and the database (which contains finalized blocks).
```

Comment:
Not a big deal, but it would be better if this (and every other) **class** comment starts with a capital letter.